src/colmap/sfm/incremental_mapper.h

```diff
@@ -231,6 +256,12 @@ class IncrementalMapper {
   // Clear the collection of changed 3D points.
   void ClearModifiedPoints3D();
 
+  // Estimate two view geometry and checks if it is suitable for initialization.
```

Comment:
How about the following interface:
```
 std::optional<TwoViewGeometry> EstimateInitialTwoViewGeometry(const Options& options,
                                      image_t image_id1,
                                      image_t image_id2);
```

Reply:
`std::optional` is a C++17 feature. Do we want to bump the minimal version to 17? Ceres has done this recently for 2.2.0

Reply:
Ah, right. Always forget. Sorry.